components/automate-cli/pkg/verifyserver/models/populatewith_test.go

```diff
@@ -0,0 +1,432 @@
+package models
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+
+	"github.com/chef/automate/lib/config"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAppendCertsByIpToNodeCerts(t *testing.T) {
+	// Test case 1: certsByIP is not nil, ipList is not empty
+	c := &Config{}
+	certsByIP := []config.CertByIP{
+		{
+			IP:         "192.168.0.1",
+			PrivateKey: "private_key1",
+			PublicKey:  "public_key1",
+		},
+		{
+			IP:         "192.168.0.2",
+			PrivateKey: "private_key2",
+			PublicKey:  "public_key2",
+		},
+	}
+	ipList := []string{"192.168.0.3", "192.168.0.4"}
+	privateKey := "private_key"
+	publicKey := "public_key"
+	adminKey := "admin_key"
+	adminCert := "admin_cert"
+
+	c.appendCertsByIpToNodeCerts(&certsByIP, ipList, privateKey, publicKey, adminKey, adminCert)
+
+	expected := []NodeCert{
+		{
+			IP:        "192.168.0.1",
+			Key:       "private_key1",
+			Cert:      "public_key1",
+			AdminKey:  "admin_key",
+			AdminCert: "admin_cert",
+		},
+		{
+			IP:        "192.168.0.2",
+			Key:       "private_key2",
+			Cert:      "public_key2",
+			AdminKey:  "admin_key",
+			AdminCert: "admin_cert",
+		},
+		{
+			IP:        "192.168.0.3",
+			Key:       "private_key",
+			Cert:      "public_key",
+			AdminKey:  "admin_key",
+			AdminCert: "admin_cert",
+		},
+		{
+			IP:        "192.168.0.4",
+			Key:       "private_key",
+			Cert:      "public_key",
+			AdminKey:  "admin_key",
+			AdminCert: "admin_cert",
+		},
+	}
+
+	// Check if each expected element exists in the actual result
+	for _, expectedNode := range expected {
+		found := false
+		for _, actualNode := range c.Certificate.Nodes {
+			if reflect.DeepEqual(expectedNode, actualNode) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			t.Errorf("Test case 1 failed: Expected node %+v not found in the result.", expectedNode)
+		}
+	}
+
+	// Test case 2: certsByIP is nil, ipList is empty
+	c = &Config{}
+	certsByIP = nil
+	ipList = []string{""}
+	privateKey = "private_key"
+	publicKey = "public_key"
+	adminKey = "admin_key"
+	adminCert = "admin_cert"
+
+	c.appendCertsByIpToNodeCerts(&certsByIP, ipList, privateKey, publicKey, adminKey, adminCert)
+
+	expected = []NodeCert{
+		{
+			IP:        "",
+			Key:       "private_key",
+			Cert:      "public_key",
+			AdminKey:  "admin_key",
+			AdminCert: "admin_cert",
+		},
+	}
+
+	// Check if each expected element exists in the actual result
+	for _, expectedNode := range expected {
+		found := false
+		for _, actualNode := range c.Certificate.Nodes {
+			if reflect.DeepEqual(expectedNode, actualNode) {
+				found = true
+				break
+			}
+		}
```

Comment:
make this an helper method called `containsElement`

Reply:
Updated code